### PR TITLE
Document PyTorch grad accumulation and add parity tests

### PR DESCRIPTION
**Pull request by Muxas**
## Summary

Implements the investigation plan for PyTorch gradient accumulation in `torch_nntile`:

- Documents that PyTorch autograd owns grad accumulation (`AccumulateGrad`, `InputBuffer`); `add`/`add_` are the integration points — no NNGraph `get_or_create_grad` layer needed.
- Registers `zero_` on `PrivateUse1` (delegates to `fill_(0)`) so `optimizer.zero_grad(set_to_none=False)` works without CPU fallback.
- Adds parity tests for diamond fan-in, microbatch accumulation (two `backward()` calls), and `grad.zero_()`.

## Changes

- [`torch_nntile/README.md`](torch_nntile/README.md): new "Gradient accumulation" section with mechanism table and training guidance.
- [`torch_nntile/csrc/nntile_kernels.cpp`](torch_nntile/csrc/nntile_kernels.cpp): register `zero_` ATen op.
- [`torch_nntile/tests/test_grad_accumulation.py`](torch_nntile/tests/test_grad_accumulation.py): three CPU vs nntile parity tests.

## Testing

```bash
export LD_LIBRARY_PATH=$PWD/build/nntile:/opt/starpu/lib
export STARPU_SILENT=1 STARPU_FXT_TRACE=0 STARPU_WORKERS_NOBIND=1
pytest -vv torch_nntile/tests/test_grad_accumulation.py
```

All 3 tests pass locally with libnntile built (`cpu_fallback=False`).

<div><a href="https://cursor.com/agents/bc-0a6aa646-cfa6-47d6-8abf-f9af770830a1"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-0a6aa646-cfa6-47d6-8abf-f9af770830a1"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by Muxas** [comment]
@cursor review
